### ISSUE TITLE
Code freezes before validation sanity check when using DDP

### ISSUE THREAD

**Issue by notprime** [PyTorchLightning/pytorch-lightning#7336]
## 🐛 Bug

Grretings from Italy!
I recently moved to PyTorch and a friend of mine introduced me to PL.
I'm coding an autoencoder (whose architecture is still pretty simple) using a custom loss function
which works on the hidden layer output. The link below leads to the github repo: 

https://github.com/notprime/custom_autoencoder/blob/main/autoenc_torch.ipynb

I read the documentation about the Multi-GPU Training, so I used '**ddp**' as accelerator, 
and used ` gpus = -1`  to select all the gpus. 
However, when I launch the script, the code freezes there:

`GPU available: True, used: True`
`TPU available: False, using: 0 TPU cores`
`Using native 16bit precision.`
`LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1,2,3]`

I tried to wait 10-15 minutes, but nothing happened.
Instead, if I use '**dp**' as accelerator, everything works fine, and the script doesn't freeze.
The documentation says that **ddp** is preferred over **dp** because it's faster:
is there something I did wrong? I really don't know why the code stucks if I use **ddp** !

Thanks in advance!

 - PyTorch Version: 1.8.1
 - OS: Ubuntu 18.04
 - How you installed PyTorch: 'conda'
 - Python version: 3.8
 - CUDA/cuDNN version: 11.2
 - GPU models and configuration: 4 x TITAN Xp 12GB

**Comment by awaelchli** [comment]
Hi,

You can't use the ddp accelerator in notebooks. Use ddp_spawn or dp for [multi gpu training](https://pytorch-lightning.readthedocs.io/en/latest/advanced/multi_gpu.html#distributed-modes). 
We added recently a check so the user gets informed that they are using an unsupported accelerator in the notebook #5970

**Comment by notprime** [comment]
@awaelchli I tried to run the code again on PyCharm, using `accelerator = 'ddp'` and `gpus = 2`,
this time the code freezes here:
```

COMET INFO: Experiment is live on comet.ml link

CometLogger will be initialized in offline mode
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
Using native 16bit precision.
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
initializing ddp: GLOBAL_RANK: 0, MEMBER: 1/2
COMET INFO: Experiment is live on comet.ml link

CometLogger will be initialized in offline mode
Using native 16bit precision.
LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [0,1,2,3]
initializing ddp: GLOBAL_RANK: 1, MEMBER: 2/2
COMET INFO: ---------------------------
COMET INFO: Comet.ml Experiment Summary
COMET INFO: ---------------------------
COMET INFO:   Data:
COMET INFO:     display_summary_level : 1
COMET INFO:     url                   : link
COMET INFO:   Uploads:
COMET INFO:     environment details : 1
COMET INFO:     filename            : 1
COMET INFO:     installed packages  : 1
COMET INFO:     os packages         : 1
COMET INFO:     source_code         : 1
COMET INFO: ---------------------------
COMET WARNING: Empty mapping given to log_params({}); ignoring

  | Name    | Type       | Params
---------------------------------------
0 | encoder | Sequential | 1.3 M 
1 | decoder | Sequential | 1.3 M 
---------------------------------------
2.6 M     Trainable params
0         Non-trainable params
2.6 M     Total params
10.524    Total estimated model params size (MB)
```

Same thing happens if I use `gpus=-1` to use the 4 gpus.
Also, if I run `nvidia-smi` in terminal, only the first GPU is on, while the other 3 are marked as off

**Comment by awaelchli** [comment]
Is it maybe because of Comet? Have you tried turning off the logger?
